### PR TITLE
Enrich store should only update the policies via an update task.

### PR DESCRIPTION
**Pull request by martijnvg**
If delete or put policy requests would happen concurrently then updates may be missed without this change.

### CODE REVIEW

**Review by hub-cap — APPROVED** [review]
Would it make more sense to put the function last in the `updateClusterState` method so that u did not have dangling args after it `}, clusterService, handler);`? but LGTM for sure. Great catch!

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java [hunk 1]
```diff
@@ -62,13 +65,15 @@ public static void deletePolicy(String name, ClusterService clusterService, Cons
             throw new IllegalArgumentException("name is missing or empty");
         }
 
-        final Map<String, EnrichPolicy> policies = getPolicies(clusterService.state());
-        if (policies.containsKey(name) == false) {
-            throw new ResourceNotFoundException("policy [{}] not found", name);
-        }
+        updateClusterState(clusterService, handler, current -> {
+            final Map<String, EnrichPolicy> policies = getPolicies(current);
```
Comment (hub-cap) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java]: should the retrieval and throw happen outside of the task? The throw will just bubble up as an exception in the caller of the `EnrichStore` whereas before it was a thrown exception.

**Review by hub-cap — APPROVED** [review]
Sorry had one more Q :P

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java]: We have to check whether the policy exists in the update task, only here we can be certain about checks. We can also check it outside the update task, before we update the cluster state, but then we would check it twice. 

The error will be caught by ClusterService and passed down to the handler (via ClusterStateUpdateTask's onFailure() method), so I think we're good here?

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Pull request merged**